support/validate-generated-templates.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -euo pipefail
+
+# the directory of the script
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+
+# the temp directory used, within $DIR
+WORK_DIR=$(mktemp -d -p "$DIR")
```

Comment:
`-p` specifies the directory to create the temporary directory in. This was basically saying create it in the same dir as source dir but can drop that requirement, no worries.